File: app/src/main/java/my/com/toru/daggertutorial/ui/MainActivity.java
package my.com.toru.daggertutorial.ui;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import javax.inject.Inject;

import my.com.toru.daggertutorial.R;
import my.com.toru.daggertutorial.remote.Test1;
import my.com.toru.daggertutorial.remote.Test2;
import my.com.toru.daggertutorial.di.DaggerMainActivityComponent;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = MainActivity.class.getSimpleName();

    @Inject
    Test1 test;
    @Inject
    Test2 test2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        DaggerMainActivityComponent.builder().build().inject(this);

        test.test();
        test2.test();
    }
}

File: kapp/src/main/java/my/com/toru/kapp/ui/main/MainActivity.kt
package my.com.toru.kapp.ui.main

import android.os.Bundle
import dagger.android.support.DaggerAppCompatActivity
import my.com.toru.kapp.R
import my.com.toru.kapp.util.DummyUtil
import my.com.toru.kapp.util.DummyUtil2
import javax.inject.Inject

class MainActivity : DaggerAppCompatActivity() {

    @Inject
    lateinit var dummyUtil: DummyUtil

    @Inject
    lateinit var dummyUtil2: DummyUtil2

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        dummyUtil.testCode()
        dummyUtil2.testCode()

        supportFragmentManager.beginTransaction()
                .replace(R.id.main_frame, MainFragment.newInstance())
                .commit()
    }
}